### DISCUSSION TITLE
Missing orange and teal color classes

### DISCUSSION THREAD

**Discussion by TomColdenhoff** [tailwindlabs/tailwindcss#3118] [Help] [answered]
I've recently installed tailwind 2.0 for the first time in my NUXTJS application. After installation, I checked out the "http://localhost:3000/_tailwind" page and noticed that the orange and teal colors were not displaying in the pallete.

Pallete with missing colors:
<img width="1007" alt="image" src="https://user-images.githubusercontent.com/23220639/102455226-81384d00-403f-11eb-8215-1bbc5cc3b122.png">

I decided to open up the CSS file (dist/tailwind.css) where the bg/text-{color} classes were at and noticed that the orange and teal colors were not in that CSS file. But the other color classes were. 

My tailwind.config.js:
```
module.exports = {
  purge: [
      './components/**/*.{vue,js}',
      './layouts/**/*.vue',
      './pages/**/*.vue',
      './plugins/**/*.{js,ts}'
      './nuxt.config.{js,ts}',
  ],
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [],
}
```

My 'assets/css/tailwind.css':
```
@tailwind base;
@tailwind components;
@tailwind utilities;
```

Can someone tell me how I can fix those missing colors?

**Comment by adamwathan** [comment, marked as answer]
Hey! See the documentation on customizing colors: 

https://tailwindcss.com/docs/customizing-colors

It explains which colors are enabled by default and how to customize it.

**Reply by TomColdenhoff** [reply]
Thank you for your response! I will try it out right away.

**Reply by digerata** [reply]
Would be great to have example copy/paste to easily include all of the colors, if desired.

**Reply by rosiecakes** [reply]
> Would be great to have example copy/paste to easily include all of the colors, if desired.

for those that visit this answer in the future: https://blog.montoulieu.dev/post/how-to-add-colors-from-the-new-color-palette-in-tailwindcss-v2-0

**Reply by unohuim** [reply]
how does restricting colours benefit anyone?  Why not include all like before?

**Reply by adamwathan** [reply]
No website needs all of these colors enabled, we're trying to encourage people to develop an actual design system and enabling 5 different sets of grays is antithetical to that.

Enabling all of them (without using JIT mode) would also double the file size, which would cripple build tooling and browser dev tools.

We'll probably enable more of them in Tailwind 3 when JIT mode is the default but we still won't enable multiple colors that are meant to be used exclusively from each other, like "gray" and "cool gray". If anything we want to add a dozen or so _more_ disabled-by-default colors so people have a bigger variety to curate from.

**Reply by unohuim** [reply]
In development/design mode, there's no such thing as too many options.  I want to play with teal and then warmGray (and thus create my colour systems in this moment while developing) without having to reconfig/compile each time.  Only in production do these restrictions make sense.  This new process breaks all of my tailwind-specifc blade components that are dependent on specific colours.

**Reply by unohuim** [reply]
If you want to manipulate the way people behave...at least allow an option to include all colours.

**Reply by adamwathan** [reply]
There is an option, it's called "writing JavaScript":

```js
let colors = require('tailwindcss/colors')

module.exports = {
  theme: {
    colors: colors,
  }
}
```

Don't be an asshole.

**Reply by unohuim** [reply]
Yikes.  Ok, so I see my issue.  I just have to remove mode:'jit" line and everything works like normal.. and when I want hyper CSS performance (like in production), I can re-enable it.  

I must be the only developer who develops during development.  Performance during development isn't a massive priority for me, options are.

Excellent feature, though.  Didn't mean to hurt your feelings.  TW is the only css I use.

**Reply by xtfer** [reply]
@adamwathan I didn't think @unohuim was being an asshole. I thought he had valid points relating to the specific problem. Your response was rude.

The way Tailwind colours are now configured is actually pretty confusing and the documentation is not helpful if you can't get things working.

**Reply by adamwathan** [reply]
This:

> If you want to manipulate the way people behave...at least allow an option to include all colours.

...is being an asshole. I am not _manipulating_ anyone, we just designed a bunch of optional colors people might want to optionally use and stuck them at the very bottom of the documentation site as a helpful resource. And like I mentioned in my reply, there _is_ an option to include any color you want, just configure those colors in your config file.

Maybe we should remove all of the extra colors from the documentation completely? Seems like our attempt to offer something helpful is generally being perceived as more negative than positive, so if we just remove everything except the defaults maybe we will deal with less vitriol :/

**Reply by xtfer** [reply]
@adamwathan Except that the thing you told him to do is basically undocumented.

**Reply by adamwathan** [reply]
There's a whole section on it right here:

https://tailwindcss.com/docs/customizing-colors#curating-colors

The fact that this:

```js
let obj = { a: 1, b: 2, c: 3}

let obj2 = {
  values: {
    a: obj.a,
    b: obj.b,
    c: obj.c,
  }
}
```

...does the same thing as this:

```js
let obj = { a: 1, b: 2, c: 3}

let obj2 = {
  values: obj
}
```

...is not Tailwind-specific knowledge. That's just JavaScript. We don't document any JavaScript language features in the Tailwind docs.

**Reply by xtfer** [reply]
And, to be fair, you are trying to affect people's practices. You said so yourself...

> No website needs all of these colors enabled, we're trying to encourage people to develop an actual design system and enabling 5 different sets of grays is antithetical to that.

**Reply by xtfer** [reply]
> We don't document any JavaScript language features in the Tailwind docs.

Okay, you guys do you. Be obtuse and complicated. Call people who can't navigate your doc's or haven't fully understood your particular usage  assholes. Good luck with that.

**Reply by adamwathan** [reply]
People can add whatever colors they want, we limit the color palette so it doesn't crash the browser dev tools and because we wouldn't use multiple grays in our own websites. Not trying to tell other people what to do, just trying to provide sensible defaults. As mentioned a dozen times you can customize things however you want, and there is lots of documentation for that. We have done nothing here to prevent anyone from doing anything they want.

Unsubscribing from this thread now 👋

**Reply by bluebrown** [reply]
> Okay, you guys do you. Be obtuse and complicated. Call people who can't navigate your doc's or haven't fully understood your particular usage assholes. Good luck with that.

Mate, you are using his project. Be nice and grateful. 

Calling Adam manipulative is kind of an asshole move, I agree.

All his points make total sense. At the end you are the professional JavaScript developer, you should know how to navigate docs or even read source code if it has to be. And you should know how to write JavaScript ofc.

**Reply by unohuim** [reply]
> And, to be fair, you are trying to affect people's practices. You said so yourself...
> 
> > No website needs all of these colors enabled, we're trying to encourage people to develop an actual design system and enabling 5 different sets of grays is antithetical to that.

100% that's straight up manipulation if you're using a dictionary as its definition.  BTW, there's also nothing wrong with manipulation (nor did I stay I had an issue with it), and there's nothing assholish about saying someone is doing such.. but yes, "encouraging' a specific set of behaviours is manipulation.  Zero reason to be offended when someone recognizes it.

**Reply by unohuim** [reply]
> People can add whatever colors they want, we limit the color palette so it doesn't crash the browser dev tools and because we wouldn't use multiple grays in our own websites. Not trying to tell other people what to do, just trying to provide sensible defaults. As mentioned a dozen times you can customize things however you want, and there is lots of documentation for that. We have done nothing here to prevent anyone from doing anything they want.
> 
> Unsubscribing from this thread now 👋

My specific issue was that I didn't realize that TW shipped with  JIT mode turned on.  I had to learn about JIT mode and understand how it impacted my development.  For instance, I couldn't use bg-red-400 unless I actually put it in the html and recompiled the assets.  Then and only then would bg-red-400 be put into my compiled css file.  But by turning JIT off, everything worked like I anticipated.  That specific behaviour wasn't anticipated. 

Also, you're in the wrong business if hearing real life feedback about a product you develop causes you to unsubscribe.  I feel like it's your job to be in all of this, not just surround yourself with the favourable echos.

**Reply by unohuim** [reply]
> > Okay, you guys do you. Be obtuse and complicated. Call people who can't navigate your doc's or haven't fully understood your particular usage assholes. Good luck with that.
> 
> Mate, you are using his project. Be nice and grateful.
> 
> Calling Adam manipulative is kind of an asshole move, I agree.
> 
> All his points make total sense. At the end you are the professional JavaScript developer, you should know how to navigate docs or even read source code if it has to be. And you should know how to write JavaScript ofc.

Being honest isn't being obtuse and complicated.  It's just being honest.  My issue had nothing to do with navigating documentation.  It was specific to JIT mode being defaulted as on.  The documents only talks about how to turn it on, so one could assume that it ships off.  Not so.  So its behaviour was surprising.

Being nice and grateful is not a prerequisite of using projects.  It's a privilege to serve this community, not the other way around.  If you produce a product for developers, IMO, you're obligate to hear everyone and not get your panties all tied-up when people communicate real issues about your product.   My guess is Adam hasn't been told anything harsh his entire life because stating he was manipulating (when he was), shouldn't have produced the response he provided.

**Reply by bluebrown** [reply]
Being nice and grateful isn't prerequisite but it helps a lot when you want something. You see what is the outcome when you come like a duche and try to tell people what's right and wrong.  People just get pissed and ignore you. 

Maybe factual you have some points maybe not, but you hamper yourself with your presentation.

Encouragement isn't manipulation, especially not in the given context. At the end he can choose whatever defaults he likes for his project. If you don't like them, don't use the project or well change it. But apparently as smart as you wish you are, you are not smart enough to change the defaults. Instead you come and cry and are rude.

> Psychological manipulation is a type of social influence that aims to change the behavior or perception of others through indirect, deceptive, or underhanded tactics. By advancing the interests of the manipulator, often at another's expense, such methods could be considered exploitative and devious.

Again, accusing Adam of outright manipulation as per above or similar definition is just rude. And given the fact you call him that to make him do what YOU want, I feel more like you are being the  manipulative one.

> Encouragement is defined as an expression of support or approval, or is words or actions that assist or inspire someone or something.
When you praise a child and encourage him to keep trying, this is an example of encouragement.

If he encourages best practices as per above of similar definition, then it not manipulation. It's encouragement. Especially since you are free to change the defaults to your liking and even got advice from Adam how to do so.

**Reply by unohuim** [reply]
ma·nip·u·late
/məˈnipyəˌlāt/
1.handle or control (a tool, mechanism, etc.), typically in a skillful manner.
2.control or **influence** (a person or situation) cleverly, unfairly, or unscrupulously.

**Reply by unohuim** [reply]
No one said anything about "psychological" manipulation.  To manipulate is a non-emotional verb.  There's no personal judgement or negative connotation associated with merely manipulating something or someone.  Advertising does it all the time.

Fun fact:  to convince me otherwise is manipulation.  I'm manipulating right now.  Influence is the key verb here.

**Reply by unohuim** [reply]
If you spent more time understanding and less time practicing sentence-writing, you'd learn that I, in fact, did change the defaults.  My only comment (and this is super foundational to all of this) is that the JIT default wasn't in the documentation and was thusly the reason for many people's confusion, including mine.  I however, found the solution on my own, and then stated it for the others to see.  I found this thread by googling my issue, so if someone in my position did so, I'd want them to see the solution.

**Reply by unohuim** [reply]
If I had known that Adam is a sensitive little bunny, I wouldn't have said anything.  I assumed that he gives two-shits about the community he writes for, and would want to address EVERYTHING.  That would be my position.  Instead, it seems that he wants people to write code and consume his framework in a specific way(the influence part?), and wants those who wish to do it another way to struggle on their own.  Totally his choice, and I don't care.  I simply thought he did.

**Reply by unohuim** [reply]
ps.. the concept of best practices IS manipulation.  There's good and bad manipulation.

**Reply by rosiecakes** [reply]
well this was certainly the most amusing issue I've ever followed. so long folks and thanks for all the colors

**Reply by bluebrown** [reply]
How do you manipulate people in a non psychological way?  Every interaction with other people is to some extent psychological. The manipulation you have stated is the umbrella term that could also be used to describe me moving a ball or manipulation it's position in space. In this specific context we can be more precise with our definition of manipulation though. We can narrow it down to psychological manipulation as this is the only thing you can do in a conversation.

Then the word manipulation has a negative taste to it. If this is aligned with the actual meaning or not doesn't really matter. It is generally precived as something negative.

I believe that you also know this. How could you not. I don't believe that you fail to understand that telling someone they are manipulation other people doesn't make them feel good. It's just not a nice way to put it. 

And yes, I didn't read much of your posts. Again you are hempering yourself with your presentation. So much so that some people don't even bother to read all the possible garbage you may or may not have said.

**Reply by unohuim** [reply]
If you produce a tool that someone uses for a year, and then you alter that tool so that it can only be used by changing the way you previously used it, you've then manipulated the end user's behaviour.  That's manipulation in its purest form. And it's not uncommon or bad.  I don't care how you've used that word in the past, or what it deeply means to you.  Manipulation (in the context of humans) is influencing their thinking or behaviour, good or bad.

**Reply by simonswiss** [reply]
Hey @unohuim,

JIT mode is not enabled by default, it's an opt-in. I am assuming you were working on a Tailwind Play? This is the only place where we have JIT turned on, because most people use Tailwind Play for quick prototyping and... play.

On that very Tailwind Play config, we also show how to add one of the colors from the extended palette:

![image](https://user-images.githubusercontent.com/485747/127415024-beda5d97-147a-4948-8fc2-3758ee7548a8.png)

Generating a default config file _does not_ include the `mode: 'jit'` option, and that's why we only document how to "turn it on". 

To the best of my knowledge, I cannot think of anywhere else than Tailwind Play where JIT mode would be turned on by default.

---

On a personal note now – I think a lot of this thread could have been avoided by using the word "encourage" instead of "manipulate". 

Sure, their _definition_ have some overlap. But the messages they carry, in text form, without the non-verbal context (tone of voice, body language) are _very_ different.

Also: you don't _have_ to be nice and grateful, on GitHub or in real life. It's indeed not a pre-requisite. 

But being kind is a _nice_ human trait that shows empathy. And usually leads to people being kind back. And help you troubleshoot.

**Reply by unohuim** [reply]
this is what my tailwind.config.js file looks like on a fresh install.

removing the mode:'jit' option brings back my original and desired behaviour.  Leaving it as such creates the new and unanticipated behaviour.  



![image](https://user-images.githubusercontent.com/7929611/127521984-48ac80e4-d6eb-4223-9040-dbc5227008fd.png)

**Reply by unohuim** [reply]
so...installing Laravel with Jetstream (laravel's preferred authentication setup), turns jit on by default.  Installing Laravel vanilla and manually bringing in TW does not.

**Reply by unohuim** [reply]
https://laravel.com/docs/8.x/starter-kits#laravel-jetstream doesn't mention any of this hugely impactful setting here, and I think it should.  Tyler's (not Adam's) page.  

But, considering how tightly integrated they are (re: TALL stack), Adam knows about Jetstream and likely knows about JIT being turned on from a fresh --jet install.  

https://tailwindcss.com/docs/just-in-time-mode#enabling-jit-mode - this page could use a section that talks about Jetstream installs.  Maybe even just a sentence would make a massive difference.

**Comment by kevalrathod** [comment]
Hey @adamwathan  I have done with cusotm color setup though i cant see teal and orange color in my website.
My tailwind.config.js:

```
const colors = require('tailwindcss/colors')

module.exports = {
  future: {
    purgeLayersByDefault: true,
    removeDeprecatedGapUtilities: true
  },
  theme: {
   extend: {
     },
     colors: {
       gray:colors.blueGray,
       teal:colors.teal,
       orange:colors.orange,
    }
}
```

**Reply by alexanderbluhm** [reply]
@kevalrathod It seems that you use Tailwind 1.x and not 2.x.
So I would suggest upgrading if you can. There is also an official Tailwind video if you have problems with this: https://www.youtube.com/watch?v=HH1eloieu2U
If you can't upgrade you can copy the colors from here: https://github.com/tailwindlabs/tailwindcss/blob/master/colors.js#L112

**Reply by kevalrathod** [reply]
@alexanderbluhm  hey thank you for response but it's tailwind 2.0.3 and i have update the custom color as per the doc. still it's missing something

**Reply by alexanderbluhm** [reply]
@kevalrathod Then you can remove the `future` part of your config.
Did you make sure that you recompiled your css?
Do you find colors like `red-500` in your css file?

**Comment by maxzakh** [comment]
This is still an issue, I have the teal color set but it is not recognized. All other colors work.
`colors: {
                'brand-blue': '#0077C2',
                'brand-purple': '#9D3BAD',
                'brand-green': '#6ABD86',
                'brand-green-light': '#AFE9AF',
                'brand-teal': '#33AFBC',
                'grad-hero-top': '#004aad',
                'grad-hero-bottom': '#5de0e6',
                'grad-green-top': '#6ABD86',
                'grad-green-bottom': '#a0d48a',
                'grad-blue-top': '#004aad',
                'grad-blue-bottom': '#0584d4',
            `},``
